custom/enikshay/management/commands/duplicate_occurrences_and_episodes_reconciliation.py

```diff
@@ -0,0 +1,228 @@
+import csv
+
+from datetime import datetime
+from django.core.management.base import BaseCommand, CommandError
+
+from corehq.apps.hqcase.utils import bulk_update_cases
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.form_processor.models import CommCareCaseSQL
+from custom.enikshay.case_utils import (
+    CASE_TYPE_OCCURRENCE,
+    CASE_TYPE_EPISODE,
+    get_occurrence_case_from_episode)
+from custom.enikshay.const import (
+    ENROLLED_IN_PRIVATE,
+)
+
+
+DOMAIN = "enikshay"
+
+
+class Command(BaseCommand):
+    """
+    1. If an open person case has multiple open occurrence cases
+       we need to keep one which is relevant and close others
+    2. If an open occurrence case has multiple open episode cases with
+       case property is_active = "yes", we need to reconcile them
+    """
+    def add_arguments(self, parser):
+        parser.add_argument('--dry_run', action='store_true')
+
+    def handle(self, *args, **options):
+        self.dry_run = options.get('dry_run')
+        self.result_file = self.setup_result_file()
+        self.case_accessor = CaseAccessors(DOMAIN)
+        # iterate all person cases
+        for person_case_id in self._get_open_person_case_ids_to_process():
+            person_case = self.case_accessor.get_case(person_case_id)
+            if self.public_app_case(person_case):
+                open_occurrence_cases = get_open_occurrence_cases_from_person(person_case_id)
+                if len(open_occurrence_cases) > 1:
+                    # reconcile occurrence cases
+                    # also reconcile episode cases under these if needed
+                    self.reconcile_cases(open_occurrence_cases, person_case_id)
+                else:
+                    # if needed reconcile episode cases under the open occurrence case
+                    self.get_open_reconciled_episode_cases_for_occurrence(open_occurrence_cases[0].get_id)
+
+    def reconcile_cases(self, open_occurrence_cases, person_case_id):
+        """
+        For each occurrence, use the following priority order to identify which (single)
+            has an open episode case where is_active=yes, episode_type = confirmed_drtb
+                if multiple, pick first opened from relevant occurrence case
+            has an open episode case where is_active=yes, episode_type = confirmed_tb
+                if multiple, pick first opened from relevant occurrence case
+            @date_opened (first opened occurrence case)
+        """
+        open_occurrence_case_ids = [case.id for case in open_occurrence_cases]
+        # get all episode cases for all open occurrences
+        all_episode_cases = []
+        for open_occurrence_case_id in open_occurrence_case_ids:
+            all_episode_cases += self.get_open_reconciled_episode_cases_for_occurrence(open_occurrence_case_id)
```

Comment:
Here you have access to the occurrence cases already, so you could make a `{episode_id: occurrence_case}` map so later on you don't need to look up occurrence cases from the DB.